network-store-client/src/main/java/com/powsybl/network/store/client/PreloadingNetworkStoreClient.java

```diff
@@ -871,4 +912,16 @@ public void removeConfiguredBuses(UUID networkUuid, int variantNum, List<String>
         ensureCached(ResourceType.CONFIGURED_BUS, networkUuid, variantNum);
         delegate.removeConfiguredBuses(networkUuid, variantNum, configuredBusesId);
     }
+
+    @Override
+    public Optional<ExtensionAttributes> getExtensionAttributes(UUID networkUuid, int variantNum, ResourceType resourceType, String identifiableId, String extensionName) {
```

Comment:
rename to getExtension for consistency

Reply:
or better getIdentifiableExtension (like  getVoltageLevelConfiguredBuses )

Reply:
Ah the return type is not resource. Add comment explaining why

Reply:
Yes I tried to have something consistent here but I'm not convinced either...
I have two options coming in mind:
```java
    Optional<ExtensionAttributes> getIdentifiableExtension(UUID networkUuid, int variantNum, ResourceType resourceType, String identifiableId, String extensionName);

    Map<String, ExtensionAttributes> getIdentifiableExtensionCollection(UUID networkUuid, int variantNum, ResourceType resourceType, String extensionName);

    Map<String, ExtensionAttributes> getIdentifiableExtensions(UUID networkUuid, int variantNum, ResourceType resourceType, String identifiableId);

    Map<String, Map<String, ExtensionAttributes>> getIdentifiableExtensionsCollection(UUID networkUuid, int variantNum, ResourceType resourceType);
```
or
```java
    Optional<ExtensionAttributes> getIdentifiableExtension(UUID networkUuid, int variantNum, ResourceType resourceType, String identifiableId, String extensionName);

    Map<String, ExtensionAttributes> getIdentifiableExtensionByType(UUID networkUuid, int variantNum, ResourceType resourceType, String extensionName);

    Map<String, ExtensionAttributes> getIdentifiableExtensions(UUID networkUuid, int variantNum, ResourceType resourceType, String identifiableId);

    Map<String, Map<String, ExtensionAttributes>> getIdentifiableExtensionsByType(UUID networkUuid, int variantNum, ResourceType resourceType);
```

The PR implementation is:
```java
    Optional<ExtensionAttributes> getExtensionAttributes(UUID networkUuid, int variantNum, ResourceType resourceType, String identifiableId, String extensionName);

    Map<String, ExtensionAttributes> getAllExtensionsAttributesByResourceTypeAndExtensionName(UUID networkUuid, int variantNum, ResourceType resourceType, String extensionName);

    Map<String, ExtensionAttributes> getAllExtensionsAttributesByIdentifiableId(UUID networkUuid, int variantNum, ResourceType resourceType, String identifiableId);

    Map<String, Map<String, ExtensionAttributes>> getAllExtensionsAttributesByResourceType(UUID networkUuid, int variantNum, ResourceType resourceType);

```